lib/base-compiler.js

```diff
@@ -941,10 +946,14 @@ export class BaseCompiler {
     }
 
     processAsm(result, filters, options) {
-        if ((options && options.includes('-emit-llvm')) || this.llvmIr.isLlvmIr(result.asm)) {
-            return this.llvmIr.process(result.asm, filters);
+        if (filters.binary && typeof result.asm !== 'string') {
```

Comment:
So; this says "if we're in binary mode and we already parsed it (cos it's not a string), then we're done...else do what we used to do. :+1: